execsnoop-kernel/execsnoop_share.cpp

```diff
@@ -64,10 +64,15 @@ int execsnoop() {
 		return err;
 	}
 
+        pb_opts.sz = sizeof(pb_opts);
+
 main_loop:
-	pb_opts.sample_cb = handle_event;
-	pb_opts.lost_cb = handle_lost_events;
-	pb = perf_buffer__new(bpf_map__fd(obj->maps.perf_events), PERF_BUFFER_PAGES, &pb_opts);
+	pb = perf_buffer__new(bpf_map__fd(obj->maps.perf_events),
```

Comment:
I think the last argument of this function should be `nullptr`. The following image contains the old (libbpf version 0.8.1) definition `perf_buffer_opts` which is a  `size_t` or callback functions.
![0.8.1](https://user-images.githubusercontent.com/61409963/194994288-bd62d1d0-4aad-4d5b-a016-07175b837b5c.png)

Reply:
I look into libbpf source, seems like this opts just use to do a check, and not passed into `__perf_buffer__new`:

https://github.com/torvalds/linux/blob/c40e8341e3b3bb27e3a65b06b5b454626234c4f0/tools/lib/bpf/libbpf.c#L11606

this check will success when opts is nullptr:

https://github.com/torvalds/linux/blob/c40e8341e3b3bb27e3a65b06b5b454626234c4f0/tools/lib/bpf/libbpf_internal.h#L296-L300

so it will be fine weather pass opts or nullptr.

I have no idea why libbpf need this opts arg.